src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePoPCryptoProvider.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    /// <summary>
+    /// This provider is based on the RSACryptoServiceProvider container concept. 
+    /// It will create and store a pair of keys in the context of the current user. Subsequent calls to this provider will reuse this key.
+    /// </summary>
+    /// <remarks>
+    /// Key creation and storage only works on Windows. See https://stackoverflow.com/questions/41986995/implement-rsa-in-net-core/42006084 for more details.
+    /// </remarks>
+    internal class NetCorePoPCryptoProvider : IPoPCryptoProvider, IDisposable
+    {
+        internal /* internal for test only */ const int RsaKeySize = 2048;
+        internal /* internal for test only */ const string ContainerName = "com.microsoft.msal";
+        private static RSA s_SigningKey;
```

Comment:
If the class is a singleton, it acts like everything is static. Consider removing the singleton pattern and just create a new instance everytime and rely on these statics.